app/controllers/images_controller.rb

```diff
@@ -93,11 +93,10 @@ def list_images
 
   # Display matrix of images by a given user.
   def images_by_user
-    user = if params[:by_user]
-             find_or_goto_index(User, params[:by_user].to_s)
-           else
-             @user
-           end
+    user = find_obj_or_goto_index(
+      model: User, obj_id: params[:by_user].to_s,
+      index_path: images_path
+    )
```

Comment:
Changes functionality when by_user is not an existing user.
See [Comment about LocationsController](https://github.com/MushroomObserver/mushroom-observer/pull/1324/files#r1095161355).